src/nav_channel.cpp

```diff
@@ -0,0 +1,221 @@
+        
+        // props will get the prop map and store it
+        prop_map = nh.subscribe("props", 10, &NavChannel::propMapCallback, this);
+
+        // global_position/local grabs position from mavros and stores it
+        global_pos = nh.subscribe("mavros/global_position/local", 10, &NavChannel::globalPositionCallback, this);
+
+        // we publish to task_status to update us on progress of task
+        task_status = nh.advertise<nav_channel::TaskStatus>("task_status", 10);
+
+        // we publish to task_goal_position to send the boat to a waypoint
+        task_goal_position = nh.advertise<nav_channel::TaskGoalPosition>("task_goal_position", 10);
+    }
+
+    void spin() {
+        ros::Rate rate(10);
+        while (ros::ok()) {
+            ros::spinOnce();
+            rate.sleep();
+        }
+    }
+
+    void setDestination(geometry_msgs::PoseStamped midpoint) {
+        // sets goal_pos and then publishes it
+        goal_pos.goal_pose = midpoint;
+        // will set orientation to (0,0,0,0), so probably want to grab current
+        // orientation and set that to new orientation
+
+        ROS_INFO_STREAM("Midpoint set at " << midpoint.pose.position.x << ","<< midpoint.pose.position.y << "," << midpoint.pose.position.z);
+
+        task_goal_position.publish(goal_pos);
+    }
+
+    geometry_msgs::PoseStamped findMidpoint() {
+        ROS_INFO("in findMidpoint.");
+        // validates positions of props, then calculates midpoint, and returns
+        // it as PoseStamped
+        geometry_msgs::PoseStamped midpoint;
+
+        bool red = false;   // use two booleans to determine if props exist
+        bool green = false; // add math to determine if valid postions
+        nav_channel::Prop red_prop;     // again fake props, change later
+        nav_channel::Prop green_prop;
+
+        for (int i = 0; !red || !green; i++) {
+            if (props.props[i].prop_label == "Red Prop") {
+                red = true;             // add math in these to check
+                red_prop  = props.props[i];   // if green, is it close enough
+            }
+            if (props.props[i].prop_label == "Green Prop") {
+                green = true;
+                green_prop = props.props[i];
+            }
+        }
+        if (red && green) {
+            float red_x = red_prop.vector.x;
+            float red_y = red_prop.vector.y;
+            float red_z = red_prop.vector.z;
+            float green_x = green_prop.vector.x;
+            float green_y = green_prop.vector.y;
+            float green_z = green_prop.vector.z;
+
+            midpoint.pose.position.x = (red_x+green_x)/2;
+            midpoint.pose.position.y = (red_y+green_y)/2;
+            midpoint.pose.position.z = (red_z+green_z)/2;
+        }
+        ROS_INFO("returning midpoint.");
+
+        return midpoint;
+    }
+
+    bool travelling() {
+        ROS_INFO("travelling...");
```

Comment:
Any ROS INFO messages that will be printed repeatedly to the terminal should be changed to ROS_DEBUG so we can limit them on the terminal when we want.